playbooks/deploy_infrastructure.yml

```diff
@@ -39,3 +39,19 @@
     - name: Reboot if there were kernel changes
       ansible.builtin.reboot:
       when: reboot.rc != 0
+
+- name: Prepare AAP install
+  hosts: installer
+  gather_facts: false
+  become: false
+  vars:
+    aap_controller_hosts: "{{ groups.controller_private }}"
+    aap_ee_hosts: "{{ groups.execution_private | default('') }}"
+    aap_hub_hosts: "{{ groups.hub_private | default('') }}"
+    aap_eda_hosts: "{{ groups.eda_private | default('') }}"
+    aap_eda_allowed_hostnames: "{{ (groups.eda_public | default([])) | join(',') }}"
+    aap_db_host: "{{ hostvars.localhost.result.response[0].properties.fullyQualifiedDomainName }}"
+  tasks:
+    - name: Create inventory
+      ansible.builtin.include_role:
+        name: lab.azure_deployment.prepare_aap
```

Comment:
A separate role for this seems a little overkill for 2 tasks, 1 template, and a couple of vars. Could this go into the infrastructure role?

Reply:
I tried but couldn't get hosts work for task. Not sure how to specify hosts for task?

Reply:
https://github.com/ansible-content-lab/azure_ansible_deployment/blob/main/roles/infrastructure/tasks/main.yml#L40 is an example